canary/CanaryUtil.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2010-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "CanaryUtil.h"
+#ifdef unix
+#    include <sys/utsname.h>
+#endif
+
+using namespace Aws::Crt;
+
+std::string CanaryUtil::GetPlatformName()
+{
+#ifdef WIN32
+    return "windows";
+#elif unix
```

Comment:
where is ```unix``` coming from? Why not use any of the C common platform macros, or have CMake define something for you?

Reply:
It's been a few months since I wrote this part, but if I recall correctly, unix is compiler defined for all unix family OS's, and the uname sorts out which one we're actually on.  I didn't realize C common had platform macros to make this easier, or that I could use CMake for this.  I'll try to look at both!